### PR TITLE
Remove phase count concept

### PR DESCRIPTION
**Pull request by yngve-sk**
🌚

### CODE REVIEW

File: src/ert/run_models/base_run_model.py [hunk 1]
```diff
@@ -336,24 +334,26 @@ def run_experiment(
     ) -> None:
         raise NotImplementedError("Method must be implemented by inheritors!")
 
-    def phaseCount(self) -> int:
-        return self._phase_count
+    def totalIterationsInExperiment(self) -> int:
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:337]: Suggest removing these functions and just operate directly on the property as there is no validation on the setter

File: src/ert/run_models/base_run_model.py [hunk 2]
```diff
@@ -336,24 +334,26 @@ def run_experiment(
...
+        self._total_iterations_in_experiment = total_iterations
 
-    def currentPhase(self) -> int:
-        return self._phase
+    def currentIteration(self) -> int:
+        return self._current_iteration
 
-    def setPhaseName(self, phase_name: str) -> None:
-        self._phase_name = phase_name
+    def setCurrentIterationLabel(self, iteration_label: str) -> None:
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:346]: Same here, can just have a property that gets updated directly

File: src/ert/run_models/multiple_data_assimilation.py [hunk 3]
```diff
@@ -81,11 +73,13 @@ def __init__(
     def run_experiment(
         self, evaluator_server_config: EvaluatorServerConfig, restart: bool = False
     ) -> None:
-        self.setPhaseCount(self.start_iteration + self.number_of_iterations)
+        # self.setTotalIterationsInExperiment(
+        #    self.start_iteration + self.number_of_iterations
+        # )
```
Comment (oyvindeide) [inline src/ert/run_models/multiple_data_assimilation.py:78]: Not needed?

File: src/ert/gui/simulation/run_dialog.py [hunk 4]
```diff
@@ -96,9 +96,11 @@ def __init__(self, snapshot_model: SnapshotModel, parent: QWidget | None) -> Non
...
             horizontal_header.setSectionResizeMode(
                 section,
-                QHeaderView.Stretch
-                if section == horizontal_header.count() - 1
-                else QHeaderView.Interactive,
+                (
+                    QHeaderView.Stretch
+                    if section == horizontal_header.count() - 1
+                    else QHeaderView.Interactive
+                ),
```
Comment (oyvindeide) [inline src/ert/gui/simulation/run_dialog.py:103]: Just cosmetic?

Comment (yngve-sk) [reply src/ert/gui/simulation/run_dialog.py:103]: Yep, no behavior change

Comment (oyvindeide) [reply src/ert/gui/simulation/run_dialog.py:103]: Suggest reverting to the original then, as it is not related to the changes in the commit

File: src/ert/run_models/evaluate_ensemble.py [hunk 5]
```diff
@@ -69,15 +69,15 @@ def run_experiment(
...
+        self._total_iterations_in_experiment = ensemble.iteration + 1
         self._evaluate_and_postprocess(
             prior_args,
             ensemble,
             evaluator_server_config,
         )
 
-        self.setPhase(phase_count, "Experiment completed.")
+        self._current_iteration_label = "Experiment completed."
+        self.setCurrentIteration(ensemble.iteration + 1)
```
Comment (oyvindeide) [inline src/ert/run_models/evaluate_ensemble.py:80]:
```suggestion
        self.current_iteration = ensemble.iteration + 1
```

File: src/ert/run_models/base_run_model.py [hunk 6]
```diff
@@ -579,11 +563,8 @@ def _build_ensemble(
     @property
     def paths(self) -> List[str]:
         run_paths = []
-        number_of_iterations = self.number_of_iterations
         active_realizations = np.where(self.active_realizations)[0]
-        for iteration in range(
-            self.start_iteration, self.start_iteration + number_of_iterations
-        ):
+        for iteration in range(self.start_iteration, self._total_iterations):
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:567]: Doesnt this change behavior?

Comment (yngve-sk) [reply src/ert/run_models/base_run_model.py:567]: | Restart   | Start iter |  number_of_iterations  | phaseCount  
| --- | --- | --- | --- |
| -  | 0  | 4  | 4  |
| 0  | 1  | 3  | 4  |
| 1  | 2  | 2  | 4  |
| 2  | 3  | 1  | 4  |

File: src/ert/run_models/base_run_model.py [hunk 7]
```diff
@@ -395,7 +392,11 @@ def send_end_event(self) -> None:
         self.send_event(
             EndEvent(
                 failed=self._failed,
-                failed_msg=self.getFailMessage(),
+                msg=(
+                    self.getFailMessage()
+                    if self._failed
+                    else self._current_iteration_label
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:398]: Looks like this is always `Experiment completed.`, can hard code that here and remove it from the run models.

**Review by oyvindeide — APPROVED** [review]
Just one small comment you can take into account if you agree, other than that, good work! Easier to follow the code this way.

Suggest keeping the final commit seperate.

File: src/ert/gui/simulation/run_dialog.py [hunk 8]
```diff
@@ -392,17 +392,15 @@ def killJobs(self) -> QMessageBox.StandardButton:
...
 
     @Slot(bool, str)
-    def _on_simulation_done(self, failed: bool, failed_msg: str) -> None:
+    def _on_simulation_done(self, failed: bool, msg: str) -> None:
         self.processing_animation.hide()
         self.kill_button.setHidden(True)
         self.restart_button.setVisible(self._run_model.has_failed_realizations())
         self.restart_button.setEnabled(self._run_model.support_restart)
-        self.update_total_progress(1.0, self._run_model.getPhaseName())
+        self.update_total_progress(1.0, msg)
```
Comment (oyvindeide) [inline src/ert/gui/simulation/run_dialog.py:400]: Or just hard code it here, and dont change the EndEvent, but both are fine

**Pull request merged**